### DISCUSSION TITLE
How Can Cortext Handle Data Gaps in Network Partition Scenarios in HA Cluster of Prometheuses?

### DISCUSSION THREAD

**Discussion by aleskxyz** [cortexproject/cortex#5633] [Q&A]
Hi,

Cortext selects a leader from the cluster of HA Prometheus to retrieve samples. Imagine a network partition situation where each Prometheus can scrape data from some instances. With the current Cortext design, only samples from the elected Prometheus will be written to long-term storage, and samples from other Prometheuses will be discarded, resulting in gaps for samples that are scraped only by the standby Prometheus.

Does Cortext have a solution for this, or can it handle this situation like Thanos, which deduplicates data at query time?

Thanks.

**Comment by alanprot** [comment]
Why do you have samples that are scraped only by the standby prometheus? I
think the idea is to have 2 prometheus instances scraping exactly the same
metrics.

The reason why we only accept 1 replica is because TSDB will reject
duplicate samples (metrics with the same timestamp for the same series with
different values).

Alan Diego

**Reply by aleskxyz** [reply]
Thanks for your reply!
As I told above, we may see this inconsistency in case of network partition.
Imagine we have 2 prometheus in 2 different racks that both of them are scraping all instances.
when internal connection between 2 racks is disrupted, then the active prometheus cannot scrape resources in the other rack but the local prometheus of that rack is still working.
Thanks

**Comment by alanprot** [comment]
So the problem is the "fail over time"?

The default value is 15 seconds its configurable: ha_tracker_update_timeout

Alan Diego
